indico/modules/events/client/js/creation.js

```diff
@@ -58,5 +76,209 @@
         }
+
+        function setLectureTimes(occurrence) {
+            startDt = moment(`${occurrence['date']} ${occurrence['time']}`, 'YYYY-MM-DD HH:mm');
+            endDt = moment(startDt).add(occurrence['duration'], 'minutes');
+        }
+
+        function initAvailabilityValues() {
+            const startDate = $('#event-creation-start_dt-date').val();
+            const startTime = $('#event-creation-start_dt-time').val();
+            const endDate = $('#event-creation-end_dt-date').val();
+            const endTime = $('#event-creation-end_dt-time').val();
+            const occurrencesVal = $('#event-creation-occurrences').val();
+            const occurrences = occurrencesVal ? JSON.parse(occurrencesVal) : null;
+
+            roomData = JSON.parse($('#event-creation-location_data').val());
+            timezone = $('#event-creation-timezone').val();
+            category = JSON.parse($('#event-creation-category').val());
+
+            if (occurrences && occurrences.length === 1) {
+                setLectureTimes(occurrences[0]);
+            } else {
+                startDt = moment(`${startDate} ${startTime}`, 'DD/MM/YYYY HH:mm');
+                endDt = moment(`${endDate} ${endTime}`, 'DD/MM/YYYY HH:mm');
+            }
+        }
+
+        function isCategoryExcluded(categoryId) {
+            return options.rbExcludedCategories.some((excludedCategory) => {
+                return (parseInt(excludedCategory.id, 10) === categoryId);
+            });
+        }
+
+        function addCalendarLink($message) {
+            const params = {
+                date: startDt.format('YYYY-MM-DD'),
+                text: roomData['room_name']
+            };
+            const url = build_url(calendarUrl, params);
+            $message.find('a').prop('href', url);
+        }
+
+        function hideAvailability(resetCheckbox) {
+            if ($currentMessage) {
+                if (resetCheckbox) {
+                    $('#availability-messages').find("input[id^='create']").prop('checked', false);
+                    $createBooking.val(false);
+                }
+                $currentMessage.hide();
+            }
+        }
+
+        function updateAvailability(resetCheckbox = false) {
+            hideAvailability(resetCheckbox);
+
+            if (!('room_id' in roomData) ||
+                !startDt.isValid() ||
+                !endDt.isValid() ||
+                !startDt.isSame(endDt, 'day') ||
+                startDt.isSameOrAfter(endDt) ||
+                isCategoryExcluded(category['id']) ||
+                timezone !== options.serverDefaultTimezone ||
+                multipleOccurrences) {
+                $createBooking.val(false);
+                return;
+            }
+
+            const roomId = roomData['room_id'];
+            previousRoomId = roomId;
+
+            const requestParams = {
+                room_id: roomId,
+                start_dt: startDt.format('YYYY-MM-DDTHH:mm:ss'),
```

Comment:
i don't think we need to include seconds here, and instead of hardcoding the format you can use `moment.HTML5_FMT.DATETIME_LOCAL`